hypothesis-python/tox.ini

```diff
@@ -166,25 +166,18 @@ commands =
     niche: bash scripts/other-tests.sh
     custom: python -bb -X dev -m pytest {posargs}
 
-[testenv:django42]
-setenv=
-    PYTHONWARNDEFAULTENCODING=1
-commands =
-    pip install django==4.2.20
-    python -bb -X dev -m tests.django.manage test tests.django {posargs}
-
-[testenv:django51]
-setenv=
-    PYTHONWARNDEFAULTENCODING=1
-commands =
-    pip install django==5.1.8
-    python -bb -X dev -m tests.django.manage test tests.django {posargs}
-
-[testenv:django52]
+[testenv:django{42,51,52,-nocontrib}]
 setenv=
     PYTHONWARNDEFAULTENCODING=1
+    DJANGO_SETTINGS_MODULE=tests.django.toys.settings.settings
+    nocontrib: DJANGO_SETTINGS_MODULE=tests.django.toys.settings.settings_no_contrib
+deps=
+    -r../requirements/test.txt
+    django42: django==4.2.22
+    django51: django==5.1.8
+    django52: django==5.2
+    django-nocontrib: django==5.2
```

Comment:
This is I think basically fine with our auto-update logic, which I wanted to check, except that there's an existing bug on master where we don't handle `major.minor` without `.patch` at the end.  Mind making that optional?